File: DataStructure/towersOfHanoi.py
# towers of hanoi
def hanoi(n, source, middle, dest, result = []):
    '''
    n: int, number of disks
    source: str, starting point (A, B or C)
    middle: str, not starting point nor destination (A, B or C)
    dest: str, destination (A, B or C)
    result: list, of process

    retuns: list, updated result
        ex. move from A to C
            move from A to B
    '''

    # base case
    if n == 1:
        result.append('move from %s to %s' % (source, dest))
        return result
    # general case
    else:
        return hanoi(n-1, source, dest, middle, []) + \
        ['move from %s to %s' % (source, dest)] +\
        hanoi(n-1, middle, source, dest, [])

n = int(raw_input('Enter the number of disks: '))
print '\n'.join(hanoi(n, 'A', 'B', 'C', ))

File: DataStructure/towersOfHanoi.c
#include <stdio.h>

// Prototype Statements
	void towers (int n, char source,
			char dest, char auxiliary);

int main (void)
{
	int numDisks;

	printf ("Please enter number of disks: ");
	scanf ("%d", &numDisks);

	printf("Start Towers of Hanoi. \n\n");

	towers (numDisks, 'A', 'C', 'B');

	printf("\nI Hope you didn't select 64 "
		"and end the world!\n");
	return 0;
}	// main

/* ================= towers ==================
*/
void towers (int n, char source,
		char dest, char auxiliary)
{
	static int step = 0;

	printf("Towers (%d, %c, %c, %c)\n",
			n, source, dest, auxiliary);

	if (n == 1)
		printf("\t\tStep %3d: Move from %c to %c\n",
			++step, source, dest);

	else
		{
		towers (n - 1, source, auxiliary, dest);
		printf("\t\tStep %3d: Move from %c to %c\n",
			++step, source, dest);
		towers (n - 1, auxiliary, dest, source);
		}	// if - else
	return;
}	// towers
